package.json

```diff
@@ -5,10 +5,11 @@
   "scripts": {
     "start": "echo THIS IS FOR LEGACY PURPOSES ONLY. && babel src -w -s -d lib --verbose --extensions \".ts,.js\"",
     "clean": "rimraf lib dist",
-    "build": "yarn clean && git rev-parse HEAD > git-revision.txt && yarn build:compile && yarn build:compile-browser && yarn build:types",
+    "build": "yarn clean && git rev-parse HEAD > git-revision.txt && yarn build:compile && yarn build:compile-browser && yarn build:minify-browser && yarn build:types",
     "build:types": "tsc --emitDeclarationOnly",
-    "build:compile": "babel src -s -d lib --verbose --extensions \".ts,.js\"",
-    "build:compile-browser": "mkdirp dist && browserify -d browser-index.js -t [ babelify --sourceMaps=inline ] > dist/browser-matrix.js",
+    "build:compile": "babel -d lib --verbose --extensions \".ts,.js\" src",
+    "build:compile-browser": "mkdirp dist && browserify -d src/browser-index.js -p [ tsify -p ./tsconfig.json ] -t [ babelify --sourceMaps=inline --presets [ @babel/preset-env @babel/preset-typescript ] ] > dist/browser-matrix.js",
```

Comment:
Hmm - sorry, but this is still spitting out the non-minified bundle with an inline source map which is now what we did before. I guess if you're worried about the file size then arguably you ought to use the minified bundle anyway, but if it works with them separate that seems nicer since you can just load the source maps if you actually want them. We previously did this with exorcist which runs after everything and takes the inline source map and splits it out into its own file.

Reply:
I would definitely argue that people should be using the minified version if they care:
![image](https://user-images.githubusercontent.com/1190097/71206894-eefe3d00-2262-11ea-9080-76ffddc29ea4.png)

I know it's not the same as before, but honestly this feels like a better way to do things.

Reply:
That misses the point though since over half of that 5.1MB is the source map. The comparison is:
```
-rw-r--r--  1 dave  staff   1.8M 19 Dec 20:32 browser-matrix-no-source-map.js
-rw-r--r--  1 dave  staff   4.9M 19 Dec 20:06 browser-matrix.js
```

...but yes, I think the only reason for not using the minified version is if you want to debug and your debugger doesn't do source maps. Not sure whether this is a case worth worrying about or not.

Why does it feel like a better way to do things? I would argue having two js files both with their source maps separate is more consistent and gives the flexibility to have source maps  if you want them.

Reply:
It's a very weak argument, but having the sourcemap inline on an already large bundle gets you debugging for free - the bundle is not a direct representation of the source, so a sourcemap is still required for it to make any sense. Also something about encouraging people to do the "right" thing and not use a 2-5mb JS file as their production build.

Reply:
It's quite tempting to rip out all of this and replace it with a webpack which gives us the 4 files for essentially the same amount of dependencies :(

browserify/babelify only supports one target, so sourcemaps are inline regardless. `exorcist` does work well enough though is starting to age a bit. It's still recommended by browserify though, so I guess we'll use it.

Reply:
should be done in https://github.com/matrix-org/matrix-js-sdk/pull/1127/commits/975368de8f6f75f31529cab62903e8b3a79640e1

result is:
![image](https://user-images.githubusercontent.com/1190097/71209032-a4cb8a80-2267-11ea-86c2-7a80f176441e.png)

Reply:
cool, thanks